src/components/PendingTransactionsSnackbar/index.js

```diff
@@ -160,9 +90,79 @@ export default class PendingTransactionsSnackbar extends Component {
         return undefined;
       }
     }
-  };
+  }
 
-  onCloseClicked = () => {
-    this.props.togglePendingTxsSnackbar(false);
-  };
+  componentDidMount() {
+    this.props.store.pendingTxsSnackbar.init();
+  }
+
+  render() {
+    const {
+      isVisible,
+      count,
+      pendingCreateEvents,
+      pendingBets,
+      pendingSetResults,
+      pendingVotes,
+      pendingFinalizeResults,
+      pendingWithdraws,
+      pendingTransfers,
+      pendingResetApproves,
+    } = this.props.store.pendingTxsSnackbar;
+    const { classes, intl } = this.props;
+
+    const pendingCounts = {
+      [TransactionType.CreateEvent]: pendingCreateEvents,
+      [TransactionType.Bet]: pendingBets,
+      [TransactionType.SetResult]: pendingSetResults,
+      [TransactionType.Vote]: pendingVotes,
+      [TransactionType.FinalizeResult]: pendingFinalizeResults,
+      [TransactionType.Withdraw]: pendingWithdraws,
+      [TransactionType.Transfer]: pendingTransfers,
+      [TransactionType.ResetApprove]: pendingResetApproves,
+    };
+
+    if (count === 0) {
+      return null;
+    }
+
+    return (
+      <Snackbar
+        className={classes.snackbar}
+        open={isVisible}
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+        message={
+          <Grid container>
+            <Grid item xs={11}>
+              <Typography variant="caption" className={classes.totalCountText}>
+                <FormattedMessage
+                  id="pendingTxsSnackbar.youHaveXPendingTransactions"
+                  defaultMessage="You have {numOfTxs} pending transaction(s)."
+                  values={{ numOfTxs: count }}
+                />
+              </Typography>
+              {
+                Object.keys(pendingCounts).map((key) => {
+                  const amount = pendingCounts[key].length;
+                  if (amount > 0) {
+                    return <Typography variant="caption" key={key}>{`${this.getEventName(key)}: ${amount}`}</Typography>;
+                  }
+                  return null;
+                })
+              }
+              <Typography variant="caption" className={classes.balanceExplanation}>
+                {`* ${intl.formatMessage(messages.balanceExplanation)}`}
+              </Typography>
+            </Grid>
+            <Grid item xs={1}>
+              <i
+                className={cx(classes.closeIcon, 'icon iconfont icon-ic_close')}
+                onClick={() => this.props.store.pendingTxsSnackbar.isVisible = false}
```

Comment:
we should be doing `isVisible` not `this.props.store.pendingTxsSnackbar.isVisible`. You're already destructuring